src/overlays/actors/ovl_En_Zo/z_en_zo.c

```diff
@@ -15,17 +15,290 @@ void EnZo_Destroy(Actor* thisx, GlobalContext* globalCtx);
+extern Gfx D_0600C028[];
+extern FlexSkeletonHeader D_0600BFA8;
+
+extern AnimationHeader D_06002FE8;
+extern AnimationHeader D_06002F10;
+extern AnimationHeader D_0600219C;
+extern AnimationHeader D_06000598;
+extern AnimationHeader D_06000D48;
+
+// Surface Ripples
+void EnZo_Ripple(EnZo* this, Vec3f* pos, f32 scale, f32 targetScale, u8 alpha) {
+    EnZoEffect* effect;
+    Vec3f vec = { 0.0f, 0.0f, 0.0f };
+    s16 i;
+
+    effect = this->effects;
+    for (i = 0; i < ARRAY_COUNT(this->effects); i++) {
+        if (effect->type == 0) {
+            effect->type = 1;
+            effect->pos = *pos;
+            effect->scale = scale;
+            effect->targetScale = targetScale;
+            effect->color.a = alpha;
+            break;
+        }
+        effect++;
+    }
+}
+
+// Bubbles
+void EnZo_Bubble(EnZo* this, Vec3f* pos) {
+    EnZoEffect* effect;
+    Vec3f vec = { 0.0f, 0.0f, 0.0f };
+    Vec3f vel = { 0.0f, 1.0f, 0.0f };
+    s16 i;
+    f32 waterSurface;
+
+    effect = this->effects;
+    for (i = 0; i < ARRAY_COUNT(this->effects); i++) {
+        if (1) {}
+        if (effect->type == 0) {
+            waterSurface = this->actor.posRot.pos.y + this->actor.waterY;
+            if (!(waterSurface <= pos->y)) {
+                effect->type = 3;
+                effect->pos = *pos;
+                effect->vec = *pos;
+                effect->vel = vel;
+                effect->scale = ((Math_Rand_ZeroOne() - 0.5f) * 0.02f) + 0.12f;
+                break;
+            }
+        } 
+        effect++;
+    }
+}
+
+// Splashes
+void EnZo_Splash(EnZo* this, Vec3f* pos, Vec3f* vel, f32 scale) {
+    EnZoEffect* effect;
+    Vec3f accel = { 0.0f, -1.0f, 0.0f };
+    s16 i;
+
+    effect = this->effects;
+    for (i = 0; i < ARRAY_COUNT(this->effects); i++) {
+        if (1) {}
+        if (effect->type != 2) {
+            effect->type = 2;
+            effect->pos = *pos;
+            effect->vec = accel;
+            effect->vel = *vel;
+            effect->color.a = (Math_Rand_ZeroOne() * 100.0f) + 100.0f;
+            effect->scale = scale;
+            break;
+        } 
+        effect++;
+    }
+}
+
+void EnZo_Ripple_Fade(EnZo* this) {
+    EnZoEffect* effect = this->effects;
+    s16 i;
+
+    for (i = 0; i < ARRAY_COUNT(this->effects); i++) {
+        if (effect->type == 1) {
+            Math_SmoothScaleMaxF(&effect->scale, effect->targetScale, 0.2f, 0.8f);
+            if (effect->color.a > 20) {
+                effect->color.a -= 20;
+            } else {
+                effect->color.a = 0;
+            }
+
+            if (effect->color.a == 0) {
+                effect->type = 0;
+            }
+        }
+        effect++;
+    }
+}
+
+// Bubbles turn into ripples when they reach the surface
+void EnZo_Bubble_CreateRipple(EnZo* this) {
+    EnZoEffect* effect;
+    f32 waterSurface;
+    s16 i;
+
+    effect = this->effects;
+    for (i = 0; i < ARRAY_COUNT(this->effects); i++) {
+        if (effect->type == 3) {
```

Comment:
do feel free to create an enum for effect type btw, if you know what they all are and want to do so.
since I dont think you would need to use it externally you can define it here toward the top of the c file near the other defines